extensions/panache/hibernate-orm-panache-kotlin/deployment/src/main/java/io/quarkus/hibernate/orm/panache/kotlin/deployment/KotlinPanacheResourceProcessor.java

```diff
@@ -0,0 +1,153 @@
+package io.quarkus.hibernate.orm.panache.kotlin.deployment;
+
+import static org.jboss.jandex.DotName.createSimple;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import javax.persistence.EntityManager;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.Type;
+import org.jetbrains.annotations.NotNull;
+
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
+import io.quarkus.arc.processor.BeanInfo;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationIndexBuildItem;
+import io.quarkus.deployment.builditem.BytecodeTransformerBuildItem;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.hibernate.orm.deployment.AdditionalJpaModelBuildItem;
+import io.quarkus.hibernate.orm.deployment.HibernateEnhancersRegisteredBuildItem;
+import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion;
+import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity;
+import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntityBase;
+import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepository;
+import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepositoryBase;
+import io.quarkus.hibernate.orm.panache.kotlin.runtime.JpaOperations;
+import io.quarkus.panache.common.deployment.EntityField;
+import io.quarkus.panache.common.deployment.EntityModel;
+import io.quarkus.panache.common.deployment.MetamodelInfo;
+import io.quarkus.panache.common.deployment.PanacheFieldAccessEnhancer;
+import io.quarkus.panache.common.deployment.PanacheRepositoryEnhancer;
+
+public final class KotlinPanacheResourceProcessor {
+
+    private static final Set<DotName> UNREMOVABLE_BEANS = Collections.singleton(
```

Comment:
This is odd. Why does the bean need to be unremovable? I mean what special thing does Kotlin Panache do that needs to make the bean unremovable when it previously wasn't required?

Reply:
That was copied from the java side.  I honestly don't know what it's there for but assumed @FroMage knew what he was doing when that went in.

Reply:
If it's in the Java side, then OK, let's keep it :)